src/params/params.js

```diff
@@ -116,6 +116,35 @@ function fetch(name, env) {
   }
 }
 
+function findCoroutine(predicate, coroutine) {
+  if (predicate(coroutine)) {
+    return coroutine;
+  } else if (_.has(coroutine, 'coroutine')) {
+    return findCoroutine(predicate, coroutine.coroutine);
```

Comment:
- I wonder if we should rename `.coroutine` to `.parentCoroutine`, or something like that?
- I also wonder if there are any circumstances where `findCoroutine` might run into an infinite loop (with two coroutines having each other as parent).